### PR TITLE
[FLINK-4521][web frontend] Fix "Submit new Job" panel in development mode

### PR DESCRIPTION
**Pull request by mushketyk**
Submit panel was completely empty in the development mode. This happened because client-side code sent AJAX requests to a development HTTP server (`localhost:3000`) and not to the the JobServer (`localhost:8081`) and development server rejected all AJAX requests.

To fix the issue I did the following:
- Updated `server.js` to pass both GET and POST requests to the JobServer (`localhost:8081`)
- Changed all AJAX requests to use development HTTP server
- Added a helper function to build a correct URL for both dev and prod modes

Also, it seems that CoffeeScript and JavaScript code were out of sync in the `master`, so this PR contains a significant number of changes to `index.js`.
- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)
- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added
- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by StephanEwen** [comment]
@rmetzger and @iampeter 
What do you think about this pull request? Does this fit with the current Web UI efforts?

File: flink-runtime-web/web-dashboard/app/scripts/index.coffee [hunk 1]
```diff
@@ -30,10 +30,15 @@ angular.module('flinkApp', ['ui.router', 'angularMoment'])
 
 .value 'flinkConfig', {
   jobServer: ''
-# jobServer: 'http://localhost:8081/'
+# jobServer: 'http://localhost:3000/job-server/'
   "refresh-interval": 10000
 }
 
+.factory 'utils', (flinkConfig) ->
+  @.jobServerUrl = (path) -> flinkConfig.jobServer + path
```
Comment (iampeter) [inline flink-runtime-web/web-dashboard/app/scripts/index.coffee]: Minor thing - we have `@`, not `@.` everywhere else.

**Comment by mushketyk** [comment]
Hi @iampeter 
Thank you for your review and sorry for the long delay.
I'll update the code in the next few days.

**Comment by mushketyk** [comment]
Hi @iampeter 
I've fixed the code according to your review.
Could you please review it again?

Best regards,
Ivan.

**Comment by iampeter** [comment]
@mushketyk looks good, thanks.

**Comment by mushketyk** [comment]
Hi @iampeter 
Thank you for your review.
Do I need to rebase this PR to merge it?

**Comment by iampeter** [comment]
@mushketyk I suppose so, right @rmetzger ?

**Comment by mushketyk** [comment]
@rmetzger @iampeter  I've rebased my PR on top of mainline.

**Comment by uce** [comment]
Sorry for the long delay with this one. The jar request URL has been fixed in another PR by now. This leaves the refactoring with the util instead of manually concatenating the URL. Should we change this @iampeter? I like it, but we would have to rebase this yet again as new requests have been added in the mean time. Sorry again for the long delay.

**Comment by mushketyk** [comment]
Hi @uce,

No worries about the long delay. Just write here if you decide to merge it and I'll rebase it.

Best regards,
Ivan.

**Comment by mushketyk** [comment]
Hi @uce 

I tried the latest version in the `master` branch and it seems that submit does not work there.

screen shot 2017-01-29 at 22 01 33

I rebased my PR on top and tested it and it seems to work fine. 
Are you sure that all issues with submit were fixed in current version?

**Comment by aljoscha** [comment]
We have a new web dashboard now.

I'm closing this as "Abandoned", since there is no more activity and the code base has moved on quite a bit. Please re-open this if you feel otherwise and work should continue.

**Pull request closed without merging**